libnmstate/state.py

```diff
@@ -935,3 +947,52 @@ def _remove_route_rule_default_values(rule):
     if rule.get(RouteRule.ROUTE_TABLE) == RouteRule.USE_DEFAULT_ROUTE_TABLE:
         del rule[RouteRule.ROUTE_TABLE]
     return rule
+
+
+def merge_linux_bridge_ports(iface_state, other_iface_state):
+    """
+    Given a linux bridge desired state, and it's current state, merges
+    those together.
+
+    This extension of the interface merging mechanism simplifies the user's
+    life in scenarios where the user wants to partially update the bridge's
+    configuration - e.g. update only the bridge's port STP configuration -
+    since it enables the user to simply specify the updated values rather
+    than the full current state + the updated value.
+    """
+    ports = iface_state.get(LinuxBridge.CONFIG_SUBTREE, {}).get(
```

Comment:
I'm opening a separated thread to discuss this merge algorithm (instead of continuing this in the  thread on line 998).

I will ask it, because I may be missing here something basic:
Why can't we follow the original interface merging pattern? What is different here?
I mean, the original interface merging has identify the same interfaces based on the names, and for those a recursive merge has been applied. Sets or lists got overwritten but empty dicts have not overwritten.
Seems to me, like the only fix needed is in the `dict_update` implementation, such that a an empty dict (val == {}) will result in overwriting the origin value. Then use it here instead of repeating a similar algorithm.

I have not tried this out yet, will try to do so and update here.